kuma/javascript/src/active-banner.jsx

```diff
@@ -182,86 +179,80 @@ function Banner(props: BannerProps) {
     );
 }
 
-type BannersProps = { banners?: Array<BannerProps> };
+export const DEVELOPER_NEEDS_ID = 'developer_needs';
+export const SUBSCRIPTION_ID = 'subscription_banner';
 
-export default function Banners(props: BannersProps) {
+const BANNER_IDS = [DEVELOPER_NEEDS_ID, SUBSCRIPTION_ID];
+
+export default function ActiveBanner() {
     const userData = useContext(UserProvider.context);
     const locale = getLocale();
 
-    // This is the data structure that defines the default list of banners
-    // that we need to consider displaying. To change the set of banners
-    // displayed on MDN pages, just edit this data structure.
-    //
-    // It is defined here inside of the component so that the
-    // gettext() calls are made after our string catalog is ready. But
-    // we do it inside a useMemo() function so that the localization
-    // only happens once.
-    const defaultBanners = useMemo(
-        () => [
-            {
-                id: 'developer_needs',
-                classname: 'developer-needs',
-                title: gettext('MDN Survey'),
-                copy: gettext(
-                    'Help us understand the top 10 needs of Web developers and designers.'
-                ),
-                cta: gettext('Take the survey'),
-                url:
-                    'https://qsurvey.mozilla.com/s3/Developer-Needs-Assessment-2019',
-                newWindow: true
-            },
-            {
-                id: 'subscription_banner',
-                classname: 'mdn-subscriptions',
-                title: gettext('Become a monthly supporter'),
-                copy: (
-                    <Interpolated
-                        id={gettext(
-                            'Support MDN with a $5 monthly subscription <learnMore/>.'
-                        )}
-                        learnMore={
-                            <a href={`/${locale}/payments/`}>
-                                {gettext('Learn more')}
-                            </a>
-                        }
-                    />
-                ),
-                cta: gettext('Subscribe'),
-                // The userData context isn't available yet so
-                // this is set when the banner is used.
-                url: '',
-                authenticated: true,
-                embargoDays: 7
-            }
-        ],
-        [locale] // This tells useMemo() to only compute this once.
-    );
+    if (!userData) {
```

Comment:
Why this?

Reply:
Wait! I get it now. The `userData` is the result of `whoami` which doesn't necessarily require that the user is authenticated. It's just a shortcut for checking if we know about the waffle flags yet.